src/bootstrap/bootstrap.py

```diff
@@ -1379,12 +1375,28 @@ def main():
     args = parse_args(sys.argv)
     help_triggered = args.help or len(sys.argv) == 1
 
-    # If the user is asking for help, let them know that the whole download-and-build
+    # Root help (e.g., x.py --help) prints help from the saved file to save the time
+    if help_triggered and sys.argv[1] in ["-h", "--help"]:
```

Comment:
I'm confused by the second part of this condition. I would have expected `if args.help and len(sys.argv == 1)` here. We'd also likely want to remove `help_triggered` since it can be just `args.help` now below.

Reply:
I modified it in https://github.com/rust-lang/rust/pull/146692/commits/accb1fbfd77d9a20c05d18159337bb39f079f4e2.

There is three patterns which bootstrap prints the top-level help:
1. User calls no argument (`./x`)
2. User calls `./x help` (bootstrap.py converts it as `./x -h` internally)
3. User uses --help or -h option in the first argument -> In current master branch, `./x --help test` also prints the top-level help, not subcommand `run`'s help.

So the appropriate condition is `if len(sys.argv) == 1 or sys.argv[1] in ["-h", "--help"]`, I think.

> We'd also likely want to remove help_triggered since it can be just args.help now below.

You're right. I removed the variable.